src/gencost/waterfall.py

```diff
@@ -2351,114 +2417,9 @@ def valid_generation(df_):
-                }
-                | {
-                    f"{k}_fraction": Column(float, Check.in_range(0.0, 1.0))
-                    for k in fuels
-                }
-                | {k: Column(float, Check.in_range(0.0, 1.0)) for k in techs}
-                | {
-                    k: Column(float, Check.gt(0.0), nullable=True)
-                    for k in ("capex", "real_opex")
-                }
-                # not used in regression
-                | {
-                    "age_in_report_year": Column(float),
-                    "age_in_current_year": Column(float, Check.in_range(0.0, 2e3)),
-                    "parasitic_load_pct": Column(float),
-                    "camd_capacity_mw": Column(float, Check.in_range(0.0, 1e4)),
-                    "gross_generation_mwh": Column(float, Check.ge(0.0)),
-                    "gross_hr": Column(float, Check.ge(0.0), nullable=True),
-                    "heat_in_mmbtu": Column(float, Check.ge(0.0)),
-                    "net_generation_mwh": Column(float),
-                    "net_cf": Column(float, nullable=True),
-                    "arc": Column(float, nullable=True),
-                    "inflator_to_2021": Column(float),
-                    "fuel_starts": Column(int, Check.ge(0)),
-                    "opex": Column(float, Check.ge(0.0), nullable=True),
-                    "real_capex": Column(float, Check.ge(0.0), nullable=True),
-                    "opex_per_kw": Column(float, Check.ge(0.0), nullable=True),
-                    "capex_per_kw": Column(float, Check.ge(0.0), nullable=True),
-                }
-                | {
-                    f"{k}_mmbtu": Column(float, Check.ge(0.0), nullable=True)
-                    for k in fuels
-                }
-                | {f"{k}_net_mwh": Column(float, nullable=True) for k in fuels}
-                | {
-                    f"{k}_gross_mwh": Column(float, Check.ge(0.0), nullable=True)
-                    for k in fuels
-                }
-                | {f"{k}_gross_cf": Column(float, Check.ge(0.0)) for k in fuels}
-            )
-
-            def gross_ge_net(df_):
-                return df_.gross_generation_mwh >= df_.net_generation_mwh
-
-            def x_gen_allocation(df_, kind):
-                return pd.Series(
-                    np.isclose(
-                        df_.filter(like=f"_{kind}_mwh").sum(axis=1),
-                        df_[f"{kind}_generation_mwh"],
-                        rtol=2e-2,
-                    ),
-                    index=df_.index,
-                )
-
-            def valid_generation(df_):
-                hrs = np.where(df_.report_date.dt.is_leap_year, 8784, 8760)
-                return df_.net_generation_mwh <= df_.capacity_mw * hrs * 1.3
-
             schema = pa.DataFrameSchema(
                 # columns=core_columns,
-                columns=merge_all_columns,
+                columns=core_columns | merge_all_columns,
```

Comment:
@arengel — I redid the pandera schema set up so that the core column dictionary has the columns in common with get_exa_by_gen and merge_all, merge_column dictionary has only the columns unique to merge_all aside from what's in the core column dict, so on and so forth..

I then combined both dictionaries as seen above. I'm getting errors indicating that combining both dictionaries in this way still might not work — when I try to run merge_all, I'm getting an error that "report_date" is missing from the df, and when I try to run get_exa_by_gen, I get notified that "generator id" is missing from the df.

This makes me think that even though the columns are in the df, pandera thinks they're not there because of the order of the keys..is that a thing?